Samples/SqueezeNetObjectDetection/Desktop/cpp/main.cpp

```diff
@@ -163,16 +189,33 @@ VideoFrame LoadImageFile(hstring filePath)
         auto stream = file.OpenAsync(FileAccessMode::Read).get();
         // Create the decoder from the stream
         BitmapDecoder decoder = BitmapDecoder::CreateAsync(stream).get();
-        // get the bitmap
-        SoftwareBitmap softwareBitmap = decoder.GetSoftwareBitmapAsync().get();
+
+        SoftwareBitmap softwareBitmap = NULL;
+        try
+        {
+            softwareBitmap = decoder.GetSoftwareBitmapAsync(
+                decoder.BitmapPixelFormat(),
+                decoder.BitmapAlphaMode(),
+                BitmapTransform(),
+                ExifOrientationMode::RespectExifOrientation,
+                colorManagementMode
+            ).get();
+        }
+        catch (hresult_error hr)
+        {
+            printf("    Failed to create SoftwareBitmap! Please make sure that input image is within the model's colorspace.\n");
+            printf("    %ws\n", hr.message().c_str());
+            exit(hr.code());
+        }
+
         // load a videoframe from it
         VideoFrame inputImage = VideoFrame::CreateWithSoftwareBitmap(softwareBitmap);
         // all done
         return inputImage;
     }
     catch (...)
     {
-        printf("failed to load the image file, make sure you are using fully qualified paths\r\n");
+        printf("    Failed to load the image file, make sure you are using fully qualified paths\r\n");

```

Comment:
Can you put this catch closer to the actual API that it is related to?

Reply:
Yep yep!